src/researchhub_document/permissions.py

```diff
@@ -1,29 +1,52 @@
-from utils.permissions import AuthorizationBasedPermission
-from utils.http import PATCH, POST, PUT, GET, DELETE
+from django.contrib.contenttypes.models import ContentType
+
+from hub.models import Hub
+from hypothesis.related_models.hypothesis import Hypothesis
+from paper.models import Paper
+from researchhub_access_group.constants import EDITOR
 from researchhub_document.related_models.constants.document_type import (
-    HYPOTHESIS,
     DISCUSSION,
+    HYPOTHESIS,
+    PAPER,
 )
-from hypothesis.related_models.hypothesis import Hypothesis
 from researchhub_document.models import (
     ResearchhubPost
 )
+from researchhub_document.related_models.researchhub_unified_document_model import ResearchhubUnifiedDocument
+from utils.permissions import AuthorizationBasedPermission
+
 
 class HasDocumentCensorPermission(AuthorizationBasedPermission):
     message = 'Need to be author or moderator to delete'
 
     def is_authorized(self, request, view, obj):
         if request.user.is_authenticated is False:
             return False
+        
+        document_type = obj.document_type if obj.__class__ is \
+            ResearchhubUnifiedDocument else Paper
+        doc = None
+        
+        if document_type == DISCUSSION:
+            doc = ResearchhubPost.objects.get(unified_document_id=obj.id)
+        elif document_type == HYPOTHESIS:
+            doc = Hypothesis.objects.get(unified_document_id=obj.id)
+        else:
+            doc = Paper.objects.get(id=obj.id)
 
-        model = None
-        if obj.document_type == DISCUSSION:
-            model = ResearchhubPost
-        elif obj.document_type == HYPOTHESIS:
-            model = Hypothesis
+        if (doc is None):
+            return False
 
-        doc = model.objects.get(unified_document_id=obj.id)
-        if doc.created_by_id == request.user.id or request.user.moderator:
+        requestor = request.user
+        is_requestor_appropriate_editor = is_hub_editor(
+            requestor,
+            doc.hubs,
+        )
```

Comment:
Let's make this a method on the user class, so we can just call `user.is_hub_editor()`